chain/network/src/network_protocol/mod.rs

```diff
@@ -507,7 +508,13 @@ impl PeerMessage {
 
 // TODO(#1313): Use Box
 #[derive(
-    borsh::BorshSerialize, borsh::BorshDeserialize, PartialEq, Eq, Clone, strum::IntoStaticStr,
+    borsh::BorshSerialize,
+    borsh::BorshDeserialize,
+    PartialEq,
+    Eq,
+    Clone,
+    strum::IntoStaticStr,
+    ProtocolSchema,
```

Comment:
I'm wondering if there is any shared set of derives for all the ProtocolSchema structs? Can we make it so that ProtocolSchema automatically adds those derives? Is it even a good idea? My primary goal would be to reduce the amount of boilerplate code.

Reply:
BorshSerialize and BorshDeserialize are the only candidates IMO. But sometimes we also have custom impls of these, so we would need to derive Borsh(De)serialize conditionally...